docs/testing/cypress.mdx

```diff
@@ -0,0 +1,193 @@
+
+```tsx filename="my-test.cy.ts"
+import { setupClerkTestingToken } from "@clerk/testing/cypress";
+
+it("sign up", () => {
+  setupClerkTestingToken();
+
+  cy.visit('/sign-up');
+  // ...
+});
+```
+
+</Steps>
+
+## Custom Commands
+
+The `@clerk/testing` package also provides [Cypress custom commands](https://docs.cypress.io/api/cypress-api/custom-commands) to sign in/sign out with Clerk in your Cypress tests without having to interact with the UI.
+To use these commands, you must import them in your [Cypress E2E Support file](https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests#Support-file), which by default is the `cypress/support/e2e.{js,jsx,ts,tsx}` file.
+
+Example:
+```ts {{ filename: 'cypress/support/commands.ts' }}
+/// <reference types="cypress" />
+import { addClerkCommands } from '@clerk/testing/cypress';
+addClerkCommands({ Cypress, cy });
+
+export {};
+```
+
+### `cy.clerkSignIn`
+
+The `cy.clerkSignIn` command is used to sign in a user using Clerk. This custom command supports only the following first factor strategies:
+
+- Password
+- Phone code
+- Email code
+
+Multi-factor authentication is not supported.
+
+> [!NOTE]
+> This helper internally uses the `setupClerkTestingToken()` helper, so you don't need to call it separately.
+
+#### Prerequisites
+
+- Before calling this command, you must call `cy.visit`.
+- Before using this command, navigate to a non-protected page that loads Clerk.
+
+#### Parameters
+
+This command accepts an object with the following properties:
+
+- `strategy`: The sign-in strategy. Supported strategies are `'password'`, `'phone_code'`, or `'email_code'`.
+    - `password`: The command will sign in the user using the provided password and identifier.
+    - `phone_code`: You must have a user with a test phone number as an identifier (e.g., `+15555550100`).
+    - `email_code`: You must have a user with a test email as an identifier (e.g., `your_email+clerk_test@example.com`).
+- `identifier`: The user's identifier. This could be a username, a phone number, or an email.
+- `password`: The user's password. This is required only if the `strategy` is set to `'password'`.
+
+
+#### Example
+
+The following example demonstrates how to use the `cy.clerkSignIn()` command in a test to sign in a user.
+
+```ts {{ filename: 'cypress/e2e/my-test.cy.ts'}}
+it('sign in', () => {
+  cy.visit(`/`);
+  cy.clerkSignIn({ strategy: 'phone_code', identifier: '+15555550100' });
+  cy.visit('/protected');
+  // user is signed in from here on
+});
+```
+
+### `cy.clerkSignOut`
+
+The `cy.clerkSignOut` command is used to sign out the current user using Clerk.
+
+#### Prerequisites
+
+- Before calling this command, yu must call `cy.visit`.
+- Before using this command, navigate to a page that loads Clerk.
+
+#### Parameters
+
+- `signOutOptions`: A `SignOutOptions` object.
```

Comment:
I had left a note with a comment before, maybe you missed it - but where is the `SignOutOptions` object? Is there a reference link for it, or do we need to provide it here.

Reply:
I missed this one. I added a link to it, also marked it as optional